test/topology_custom/test_mv_admission_control.py

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+
+from test.topology.conftest import skip_mode
+from test.pylib.util import wait_for_view
+from test.topology_experimental_raft.test_mv_tablets import pin_the_only_tablet, get_tablet_replicas
+
+from cassandra.cluster import ConsistencyLevel, EXEC_PROFILE_DEFAULT # type: ignore
+from cassandra.cqltypes import Int32Type # type: ignore
+from cassandra.policies import FallthroughRetryPolicy # type: ignore
+from cassandra.query import SimpleStatement, BoundStatement # type: ignore
+
+logger = logging.getLogger(__name__)
+
+# In the test, we create a table and perform a pair of writes to it. The
+# second write should fail due to admission control. We check that this
+# is indeed the error thrown.
+@pytest.mark.asyncio
+@skip_mode('release', "error injections aren't enabled in release mode")
+async def test_mv_admission_control_exception(manager: ManagerClient) -> None:
+    node_count = 2
+    config = {'error_injections_at_startup': ['view_update_limit', 'delay_before_remote_view_update', 'update_backlog_immediately'], 'enable_tablets': True}
+    servers = await manager.servers_add(node_count, config=config)
+    cql, hosts = await manager.get_ready_cql(servers)
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}}"
+                        "AND tablets = {'initial': 1}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, v text, PRIMARY KEY (key, c))")
+    await cql.run_async(f"CREATE MATERIALIZED VIEW ks.mv_cf_view AS SELECT * FROM ks.tab "
+                    "WHERE c IS NOT NULL and key IS NOT NULL PRIMARY KEY (c, key) ")
+    await wait_for_view(cql, 'mv_cf_view', node_count)
+
+    # Only remote updates hold on to memory, so make the update remote by pinning base and view tablets to different nodes.
+    await pin_the_only_tablet(manager, "ks", "tab", servers[0])
+    await pin_the_only_tablet(manager, "ks", "mv_cf_view", servers[1])
+
+    # Prepare the statement so that the write goes to the same shard both
+    # times (the first write will cause only the shard on which it was
+    # performed to have the updated view update backlog).
+    stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+    # To inspect the error message, we need to disable retries, which can't
+    # be done in `prepare()` or `run_async()`. Instead, we use `BoundStatement`.
+    bnd_stmt = BoundStatement(stmt, retry_policy=FallthroughRetryPolicy())
+    await cql.run_async(bnd_stmt.bind([0, 0, 240000*'a']), host=hosts[0])
+    with pytest.raises(Exception, match="View update backlog is too high"):
+        await cql.run_async(bnd_stmt.bind([0, 0, 240000*'a']), host=hosts[0])
```

Comment:
I didn't understand this. Why is the second write already too much backlog? 240KB doesn't seem so much. Did you configure the backlog to be artificially low? Or is it this low just because it's 80% of 10% of something relatively small?
If the answer to my last question is yes, it means that this test will break if we ever change the default amount of memory or default amount of memory dedicated to view backlog. That isn't nice... I think in that case it will be better to have this test explicitly configure Scylla's memory limit, or, alternatively, have the test inquire Scylla's configuration and calculate this 240000 number from it.

Reply:
Oh, or maybe the view_update_limit injection is what caused the second write to fail?
But if that's the case, why did you need that large 240KB write? Wouldn't the string 'a' fail just the same?

Reply:
Right, it's due to the `view_update_limit` injection - it sets the view backlog limit to 250000 bytes. It would probably be clearer to show that just any following write fails (like a write with a single 'a'), so I'm replacing the second write with a tiny one in the rebase